src/lib/utils/router.js

```diff
@@ -5,6 +5,11 @@ let recentActiveUrl; // current URL not including hash
 // async, and most browsers will move the scroll position automatically for us, even on old content.
 // Instead, we call `scrollOnFrame` when the async load helper is done.
 window.history.scrollRestoration = "manual";
+window.addEventListener("beforeunload", (e) => {
```

Comment:
I'm not sure adding an unconditional `beforeunload` handler is the right solution here (as it can [break bfcache in browsers](https://developers.google.com/web/updates/2018/07/page-lifecycle-api#the-beforeunload-event)). Would it work if you used the `pagehide` event?

Reply:
TIL `pagehide`. Yes this fixes it.